Paperless-ng cannot add documents

### Describe the bug

Whenever I try to add a document using the web ui, it uploads and then waits forever to finish processing. Looking at the log in the same UI, I see the following lines:

```console
...
[2021-08-19 15:24:37,558] [ERROR] [paperless.consumer] Cannot consume /tmp/paperless/paperless-upload-41faim54: File not found.
[2021-08-19 15:46:09,759] [ERROR] [paperless.consumer] Cannot consume /tmp/paperless/paperless-upload-b4jpxtsc: File not found.
...
```

And when doing `systemctl status paperless-ng-server.service` I see that it fails with `documents.consumer.ConsumerError: 2021-07.pdf: Cannot consume /tmp/paperless/paperless-upload-b4jpxtsc: File not found.`.

This is clearly due to the `privateTmp = true` in the services, instead we should share `/tmp` between the `paperless-ng-web` and `paperless-ng-server` services (and potentially the `consumer` too?).


### Steps To Reproduce
Steps to reproduce the behavior:
1. `services.paperless-ng.enable = true;`
2. upload a document
3. wait forever

### Expected behavior

The document should upload and finish processing correctly, being readable accross services.

### Notify maintainers

@Flakebi @erikarvstedt

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.94, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210802_47e96bb`
 - channels(ambroisie): `""`
 - channels(root): `"nixos-21.11pre293089.1c2986bbb80"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
- paperless-ng
```

--- a/nixos/modules/services/misc/paperless-ng.nix
+++ b/nixos/modules/services/misc/paperless-ng.nix
@@ -283,6 +283,9 @@ in
         PATH = mkForce cfg.package.path;
         PYTHONPATH = "${cfg.package.pythonPath}:${cfg.package}/lib/paperless-ng/src";
       };
+      # Allow the web interface to access the private /tmp directory of the server.
+      # This is required to support uploading files via the web interface.
+      unitConfig.JoinsNamespaceOf = "paperless-ng-server.service";
       # Bind to `paperless-ng-server` so that the web server never runs
       # during migrations
       bindsTo = [ "paperless-ng-server.service" ];

--- a/nixos/tests/paperless-ng.nix
+++ b/nixos/tests/paperless-ng.nix
@@ -25,12 +25,22 @@ import ./make-test-python.nix ({ lib, ... }: {
         # Wait until server accepts connections
         machine.wait_until_succeeds("curl -fs localhost:28981")
 
-    with subtest("Document is consumed"):
+    with subtest("Create web test doc"):
+        machine.succeed(
+            "convert -size 400x40 xc:white -font 'DejaVu-Sans' -pointsize 20 -fill black "
+            "-annotate +5+20 'hello web 16-10-2005' /tmp/webdoc.png"
+        )
+        machine.wait_until_succeeds("curl -u admin:admin -F document=@/tmp/webdoc.png -fs localhost:28981/api/documents/post_document/")
+
+    with subtest("Documents are consumed"):
         machine.wait_until_succeeds(
-            "(($(curl -u admin:admin -fs localhost:28981/api/documents/ | jq .count) == 1))"
+            "(($(curl -u admin:admin -fs localhost:28981/api/documents/ | jq .count) == 2))"
         )
         assert "2005-10-16" in machine.succeed(
             "curl -u admin:admin -fs localhost:28981/api/documents/ | jq '.results | .[0] | .created'"
+        )
+        assert "2005-10-16" in machine.succeed(
+            "curl -u admin:admin -fs localhost:28981/api/documents/ | jq '.results | .[1] | .created'"
         )
   '';
 })

